kafka-connect-s3/src/test/java/io/confluent/connect/s3/DataWriterByteArrayTest.java

```diff
@@ -100,9 +101,26 @@ public void testNoSchema() throws Exception {
     verify(sinkRecords, validOffsets, context.assignment(), ".bin");
   }
 
+  @Test
+  public void testGzipCompression() throws Exception {
+    CompressionType compressionType = CompressionType.GZIP;
+    localProps.put(S3SinkConnectorConfig.FORMAT_CLASS_CONFIG, ByteArrayFormat.class.getName());
+    localProps.put(S3SinkConnectorConfig.COMPRESSION_TYPE_CONFIG, compressionType.name);
+    setUp();
+    task = new S3SinkTask(connectorConfig, context, storage, partitioner, format, SYSTEM_TIME);
+
+    List<SinkRecord> sinkRecords = createByteArrayRecordsWithoutSchema(7 * context.assignment().size(), 0, context.assignment());
+    task.put(sinkRecords);
+    task.close(context.assignment());
+    task.stop();
+
+    long[] validOffsets = {0, 3, 6};
+    verify(sinkRecords, validOffsets, context.assignment(), ".bin.gz");
+  }
+
   @Test
   public void testCustomExtensionAndLineSeparator() throws Exception {
-    String extension = ".SEPARATORjson";
+    String extension = ".customExtensionForTest";
```

Comment:
Is this a more understandable custom extension for this test?

Reply:
I think it is. Thanks!